### PR TITLE
Use `uv` to speed up pip installs & the CI in general

### PR DESCRIPTION
**Pull request by strickvl**
Uses [`uv`](https://github.com/astral-sh/uv) to speed up Python package installation in the CI.

I also removed the cache completely, so @safoinme not sure if we need the infra for that any more? (`uv` is so fast we don't need it).

### CODE REVIEW

**Comment by strickvl** [comment]
Pausing work on this until https://github.com/astral-sh/uv/issues/1526 is added / released.

File: tests/unit/utils/test_source_utils.py
Comment (strickvl) [inline tests/unit/utils/test_source_utils.py:1]: @schustmi not sure if this counts as 'fixing the test' or not. `uv` seems to install Python differently on Windows (see full discussion [here](https://github.com/astral-sh/uv/issues/2056)) which results in different behaviour when on windows for these two source utils for `defaultdict`. Any objections, or should I dive deeper into what's different?

**Comment by safoinme** [comment]
Let's start with no caching, but we will need to have it later on as part of the Azure cost optimization for these runners, it's just a question of how does uv deal with cache

**Review by avishniakov — APPROVED** [review]

File: tests/unit/utils/test_source_utils.py
Comment (strickvl) [reply tests/unit/utils/test_source_utils.py:1]: How about like this @schustmi? Unfortunately `int` doesn't have an associated file, so we can't check for that dunder attribute.

**Pull request merged**